exercises/concept/rpn-calculator/.meta/exemplar.rs

```diff
@@ -0,0 +1,47 @@
+#[derive(Debug)]
+pub enum CalculatorInput {
+    Add,
+    Subtract,
+    Multiply,
+    Divide,
+    Value(i32),
+}
+
+pub fn evaluate(inputs: &[CalculatorInput]) -> Option<i32> {
+    let mut stack = Vec::new();
+
+    for input in inputs {
+        use CalculatorInput::*;
+
+        match input {
+            Add => {
+                let rhs = stack.pop()?;
+                let lhs = stack.pop()?;
+                stack.push(lhs + rhs);
+            }
+            Subtract => {
+                let rhs = stack.pop()?;
+                let lhs = stack.pop()?;
+                stack.push(lhs - rhs);
+            }
+            Multiply => {
+                let rhs = stack.pop()?;
+                let lhs = stack.pop()?;
+                stack.push(lhs * rhs);
+            }
+            Divide => {
+                let rhs = stack.pop()?;
+                let lhs = stack.pop()?;
+                stack.push(lhs / rhs);
+            }
+            Value(value) => stack.push(*value),
+        }
+    }
+
+    let output = stack.pop();
+    if stack.is_empty() {
+        output
+    } else {
+        None
+    }
```

Comment:
```suggestion
    stack.is_empty().then(|| output)
```

Not essential, but fun: `bool` has a [`then` combinator](https://doc.rust-lang.org/std/primitive.bool.html#method.then) to simplify cases like these.

Reply:
I am a big fan of `then()`, but I didn't want to have more prerequisites. Should this be taught in `role-playing-game` (the `option` concept exercise).

Reply:
AFAIK we are allowed to do things in the exemplar that are not covered by the prerequisites. The prerequisites are what is necessary to implement a simple solution, but we aren't restricted by them. 

However, I'm not perfectly confident about my understanding of that policy. @efx what do you think?

Reply:
https://github.com/exercism/v3/blob/main/docs/concept-exercises.md#exemplar-implementation-file

> The implementation should only use language features introduced by the exercise or its prerequisites (and their prerequisites, and so on).

Looks like we shouldn't use things not previously covered.

Reply:
Thanks for finding the reference! I was wrong; let's keep it as it currently is.